naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimeUnionField.scala

```diff
@@ -48,8 +49,13 @@ object NaptimeUnionField {
         case definitions: java.util.Map[String @unchecked, String @unchecked] => definitions.asScala
       }.getOrElse(Map[String, String]())
 
-      val unionMemberKey = typedDefinitions
-        .getOrElse(subType.getUnionMemberKey, subType.getUnionMemberKey)
+      val unionMemberKey = (subType, typedDefinitions) match {
```

Comment:
it doesn't look like we actually need to match against `typedDefinitions`, since we're not really doing anything with it that we can't do oyn `typedDefinitions` itself.
I'd probably recommend only matching against subType to handle the different types for that, and always look up in typedDefinition from the different options. We could also do `getOrElse`s to kind of keep trying different things until we find a match in the typedDefinition map, or fall back to the unionMemberKey.

Let's chat more in person maybe